src/main.rs

```diff
@@ -167,12 +168,19 @@ fn main() -> Result<(), failure::Error> {
     } else if let Some(matches) = matches.subcommand_matches("generate") {
         let name = matches.value_of("name").unwrap_or("worker");
         let project_type = match matches.value_of("type") {
-            Some(s) => Some(settings::project::ProjectType::from_str(&s.to_lowercase())?),
+            Some(s) => Some(ProjectType::from_str(&s.to_lowercase())?),
             None => None,
```

Comment:
i know it's not part of this PR, but based on the deeper layers of this code, it should be possible to always return just a bare `ProjectType` rather than an `Option` here; the `generate` command should also be capable of taking this. That would get rid of the nested match below and clean this up. iirc, the default would be `webpack`

Reply:
reviewed the code again. turns out this refactor depends on the incoming solution for #315.